### PR TITLE
Add support for jar/zip format for local-path flag

### LINKED ISSUE #29: --local-path supports jar/zip
### Description of problem

There are processes/pipelines being utilized today which generate a jar/zip file artifact.

These pipelines will often pull in additional assets which aren't included in the application code itself, but which are necessary for the application function.
A subsequent next step is for the jar/zip of the app to be pushed to and run on a target environment.
Popular platforms such as Cloud Foundry support this workflow today (via the cf CLI). 

The apps plugin for the Tanzu CLI _does not_ support creating/updating/applying a workload from a local jar or zip directly.
The file must be unzipped beforehand. 
Requiring the jar be unzipped is not an expected step, it creates toil without value, and decreases efficiency.

### Proposed solution

```gherkin
Given I've got a JAR (or zipped) file containing all assets required to create a workload and I'm targeting a TAP-enabled workload cluster
When I run `tanzu apps workload create myapp -f path/to/my/myapp.jar...`
Then my workload create command exits 0 and I can validate the workload has run through the supply chain by viewing the workload details via `tanzu apps workload get...`
```

@danfein to provide UX here

### Additional context
* cc: @matthewmcnew @jigsheth57 may have additional context/thoughts on this issue
* from @scothis: 
  * It's worth mirroring the behavior of pack. https://github.com/buildpacks/pack/blob/3fc8e2220b584da76cd0227bbc211aae787e0e57/pkg/client/build.go#L596-L605

### PR DESCRIPTION
**Pull request by shashwathi**
### What this PR does / why we need it

Add support for jar/zip format for local-path flag

### Which issue(s) this PR fixes
Fixes #29 

### Describe testing done for PR
Download simple java jar file and build zip file(using `zip` cli).

### CODE REVIEW

File: pkg/commands/workload.go [hunk 1]
```diff
@@ -324,8 +324,23 @@ func (opts *WorkloadOptions) PublishLocalSource(ctx context.Context, c *cli.Conf
...
+	} else if source.IsZip(opts.LocalPath) {
+		zipContentsDir, err := source.HandleZip(opts.LocalPath)
+		if err != nil {
+			c.Errorf("Failed to extract file contents from %q. \n", opts.LocalPath)
+			return false, err
+		}
+		defer os.RemoveAll(zipContentsDir)
+		contentDir = zipContentsDir
+	} else {
+		return false, fmt.Errorf("unsupported file format %q", opts.LocalPath)
```
Comment (shashwathi) [inline pkg/commands/workload.go:339]: cc @danfein : this is the error if user provides a file which is not a dir and not a valid zip file. If you want something else, please add the msg here

File: pkg/source/archive.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	"archive/zip"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+)
+
+func HandleZip(fileName string) (string, error) {
+	tmpDir, err := ioutil.TempDir("", "")
```
Comment (rashedkvm) [inline pkg/source/archive.go:29]: issue: When is the temp directory removed?
```
	tmpDir, err := ioutil.TempDir("", "")
```

File: pkg/source/archive.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+		}
+
+		if err := srcFile.Close(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func IsZip(fileName string) bool {
```
Comment (rashedkvm) [inline pkg/source/archive.go:99]: Question: `IsZip` testing both "jar" and "zip" file types?

File: pkg/commands/workload.go [hunk 4]
```diff
@@ -324,8 +324,23 @@ func (opts *WorkloadOptions) PublishLocalSource(ctx context.Context, c *cli.Conf
...
+	var contentDir string
+	if source.IsDir(opts.LocalPath) {
+		contentDir = opts.LocalPath
+	} else if source.IsZip(opts.LocalPath) {
+		zipContentsDir, err := source.HandleZip(opts.LocalPath)
+		if err != nil {
+			c.Errorf("Failed to extract file contents from %q. \n", opts.LocalPath)
+			return false, err
+		}
+		defer os.RemoveAll(zipContentsDir)
```
Comment (warango4) [inline pkg/commands/workload.go:336]: > issue: Shouldn't the temp directory be cleaned up at some point (removed)?
@rashedkvm I think that's being done here.

**Review by warango4 — APPROVED** [review]

File: pkg/source/archive.go (hunk 2 above)
Comment (shashwathi) [reply pkg/source/archive.go:29]: So we need the temp dir for imgpkg to package up the contents. So in the called function, tmp dir is cleanedup.

Comment (scothis) [reply pkg/source/archive.go:29]: it would be better to accept the directory as an argument, and then the cleanup is clearly the callers problem

**Review by rashedkvm — APPROVED** [review]

**Pull request merged**